kustomize-fns/v1alpha1/applyresources/ApplyResources.go

```diff
@@ -0,0 +1,165 @@
+package main
+
+import (
+	"fmt"
+	log "github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	kubeTypes "k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/discovery/cached/memory"
+	"k8s.io/client-go/dynamic"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/restmapper"
+	"sigs.k8s.io/kustomize/v3/pkg/ifc"
+	"sigs.k8s.io/kustomize/v3/pkg/resmap"
+	"sigs.k8s.io/kustomize/v3/pkg/resource"
+	"sigs.k8s.io/kustomize/v3/pkg/transformers/config"
+	"sigs.k8s.io/kustomize/v3/pkg/types"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	// configurableResourcesLabel is a label added by odh dev to specify which objects are can be updated.
+	configurableResourcesLabel = "opendatahub.io/configurable"
+	// modifiedResourcesLabel is a label added by end user to specify a given resource has been updated.
+	modifiedResourcesLabel     = "opendatahub.io/modified"
+	// needsUpdateResourcesLabel is a label added by odh devs to specify that an object needs to be updated with latest
+	// changes irrespective of the modified label.
```

Comment:
@VaishnaviHire just to clarify the needs-update label is created by odh devs to specify this resource was changed during the last update?

Reply:
Yes that is correct.